src/backend/src/api/types/deployments.rs

```diff
@@ -1,50 +1,25 @@
-use std::borrow::Cow;
-
+use super::UserId;
 use candid::{CandidType, Decode, Encode};
 use ic_stable_structures::{storable::Bound, Storable};
-use serde::Deserialize;
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
 use utils::{get_time_nanos, Uuid};
 
-use super::UserId;
-
 pub type DeploymentId = Uuid;
 
-#[derive(Debug, CandidType, Deserialize, Clone, PartialEq, Eq)]
-pub enum DeploymentState {
-    Initialized,
-    DeploymentCreated,
-    LeaseCreated,
-    Active,
-    Closed,
-}
-
-impl Storable for DeploymentState {
-    fn to_bytes(&self) -> Cow<[u8]> {
-        Cow::Owned(Encode!(self).unwrap())
-    }
-
-    fn from_bytes(bytes: Cow<[u8]>) -> Self {
-        Decode!(bytes.as_ref(), Self).unwrap()
-    }
-
-    const BOUND: Bound = Bound::Unbounded;
-}
-
 #[derive(Debug, CandidType, Deserialize, Clone, PartialEq, Eq)]
 pub struct Deployment {
     sdl: String,
     user_id: UserId,
-    created_at: u64,
-    state: DeploymentState,
+    state_history: Vec<(u64, DeploymentUpdate)>,
```

Comment:
there's a `TimestampNs` time for timestamps, instead of `u64`, just for better readability